jbmc/src/java_bytecode/README.md

```diff
@@ -736,3 +736,151 @@ See also \ref string-solver-interface.
+
+This concept becomes clearer with an example.
+
+\subsection java-trace-member-assignment-example Worked example for member assignments
+
+Consider the trace for the function call to `Example.assignReferenceTypes()`,
+which assigns a value to the static field `globalReferenceTypeReferenceField`.
+
+```java
+public class Example {
+  static ReferenceTypeReferenceField globalReferenceTypeReferenceField;
+  public static void assignReferenceTypes() {
+    globalReferenceTypeReferenceField =
+        new ReferenceTypeReferenceField(new ReferenceTypePrimitiveField(5));
+}
+
+class ReferenceTypeReferenceField {
+  ReferenceTypePrimitiveField referenceField;
+  public ReferenceTypeReferenceField(ReferenceTypePrimitiveField r) {
+    referenceField = r;
+  }
+}
+
+class ReferenceTypePrimitiveField {
+  int primitiveField;
+  public ReferenceTypePrimitiveField(int i) {
+    primitiveField = i;
+  }
+}
+```
+
+There are 54 non-trivial steps in the trace to make this assignment.
+This example will give detail of the assignment steps only, filling in other
+steps with narrative.
+
+CProver start initializes the instance of the class containing the function under test, Example, to have `null` static field:
+- \ref symbol_exprt `java::ObjectReference.globalReferenceTypeReferenceField` = \ref constant_exprt with `null` value
+
+Objects are empty-initialized to build the assignment to `globalReferenceTypeReferenceField`.
+`symex_dynamic::dynamic_object1` represents `globalReferenceTypeReferenceField` and `symex_dynamic::dynamic_object2` represents `referenceField`.
+- \ref symbol_exprt `symex_dynamic::dynamic_object1` = \ref struct_exprt with `referenceField` set to `null` value
+- \ref symbol_exprt `symex_dynamic::dynamic_object2` = \ref struct_exprt with `primitiveField` set to `0`
+
+Call to the constructor of `ReferenceTypePrimitiveField`, initializing the parameters `this` and `i`
+- \ref symbol_exprt `java::ReferenceTypePrimitiveField.<init>:(I)V::this` = \ref const_exprt with \ref address_of_exprt pointing to `symex_dynamic::dynamic_object2`
+- \ref symbol_exprt `java::ReferenceTypePrimitiveField.<init>:(I)V::i` = \ref const_exprt with value `5`
+
+Call to the super constructor (Object). The `to_construct` parameter is internal
+- \ref symbol_exprt `java::java.lang.Object.<init>:()V::this` = \ref const_exprt with \ref address_of_exprt pointing to `symex_dynamic::dynamic_object2`
+- \ref symbol_exprt `java::java.lang.Object.<init>:()V::to_construct` = \ref const_exprt with \ref address_of_exprt pointing to `symex_dynamic::dynamic_object2`
+
+Setting the member of symex_dynamic::dynamic_object2, primitiveField. Note that primitives are always assigned using constant expressions with values.
+- \ref member_exprt `symex_dynamic::dynamic_object2.primitiveField` = \ref const_exprt with value `5`
+
+Call to the constructor of ReferenceTypeReferenceField, initalizing the parameters `this` and `r`
```

Comment:
⛏ `, initializing the parameters`, you've forgotten `i`

Reply:
Fixed typo, but there is no parameter i in this constructor.